packages/wallet-sdk/src/relay/Web3Request.ts

```diff
@@ -11,164 +11,128 @@ import {
 } from '../types';
 import { Web3Method } from './Web3Method';
 
-interface BaseWeb3Request<
-  Method extends Web3Method,
-  Params extends object = Record<string, unknown>,
-> {
-  method: Method;
-  params: Params;
-}
-
-export type RequestEthereumAccountsRequest = BaseWeb3Request<
-  Web3Method.requestEthereumAccounts,
-  RequestEthereumAccountsBaseParams
->;
-
-type RequestEthereumAccountsBaseParams = {
-  appName: string;
-  appLogoUrl: string | null;
-};
-
-export type ConnectAndSignInRequest = BaseWeb3Request<
-  Web3Method.connectAndSignIn,
-  RequestEthereumAccountsBaseParams & {
-    domain: string;
-    aud: string;
-    version: string;
-    type: string;
-    nonce: string;
-    iat: string;
-    chainId: string;
-    statement?: string;
-    resources?: string[];
-  }
->;
-
-export type AddEthereumChainRequest = BaseWeb3Request<
-  Web3Method.addEthereumChain,
-  {
-    chainId: string;
-    blockExplorerUrls?: string[];
-    chainName?: string;
-    iconUrls?: string[];
-    rpcUrls: string[];
-    nativeCurrency?: {
-      name: string;
-      symbol: string;
-      decimals: number;
+export type Web3Request<M extends Web3Method = Web3Method> = Extract<_Web3Request, { method: M }>;
```

Comment:
this allows to use
- `Web3Request` as it is to specify any requests with any method type
- e.g. `Web3Request<'requestEthereumAccounts'>` to confine method
- union type like e.g. `Web3Request<'signEthereumMessage' | 'submitEthereumTransaction'>`